python/snewpy/snowglobes.py

```diff
@@ -91,6 +94,22 @@ def generate_time_series(model_path, model_type, transformation_type, d, output_
     tedges = np.arange(tmin/u.s, tmax/u.s, dt/u.s)*u.s
     times = 0.5*(tedges[1:] + tedges[:-1])
 
+    # now process log data
+    if (log_bins==True):
+        log_edges = np.asarray([])
+
+        if tmax < 0 or tmin < 0:
+            raise ValueError("Cannot apply log to time windows that are less than 0")
+        tstep = math.log10(abs(tmax/tmin))/len(times)
+
+        for i in range(0,len(times)):
+            t = (tmin/u.s)*(10**(i*tstep))
+            if t < 0:
```

Comment:
Can `t` ever be less than 0 here, if we’ve already checked a few lines earlier that `tmin` is positive?